File: M3/eclipse-workspace/UF2/E07_Frases-Celebres/src/sax/Tema.java
package sax;

import java.util.ArrayList;

public class Tema {
	private String nom;
	private ArrayList<Frase> frases;
	
	public Tema() {
		this.frases = new ArrayList<Frase>();
	}
	
	public String getNom() {
		return nom;
	}
	public ArrayList<Frase> getFrases() {
		return frases;
	}
	public void setNom(String nom) {
		this.nom = nom;
	}
	public void setFrases(ArrayList<Frase> frases) {
		this.frases = frases;
	}
}

File: DAW-2/M7/eclipse-workspace/E03_Frases/src/Frases/Entity/Tema.php
<?php
namespace Frases\Entity;

use Doctrine\Common\Collections\ArrayCollection;

/**
 * Tema
 *
 * @Entity
 * @Table(name="tbl_temes")
 */
class Tema {
    /**
     * @var int
     *
     * @Id
     * @GeneratedValue(strategy="AUTO")
     * @Column(type="integer", nullable=false)
     */
    private $id;
    
    /**
     * @var string
     *
     * @column(type="string", length=50, nullable=false)
     */
    private $nombre;
    
    /**
     * @var Frase[] 
     * 
     * @ManyToMany(targetEntity="Frase", mappedBy="temes")
     * @JoinTable(name="tbl_frases_temes")
     */
    private $frases;
    
    public function __construct(){
        $this->frases = new ArrayCollection();
    }
    
    /**
     * Add Frase
     *
     * @param Frase $frase
     * @return Tema
     */
    public function addFrase(Frase $frase){
        $this->frases[] = $frase;
        return $this;
    }
    
    public function getId(){
        return $this->id;
    }
    public function getNombre(){
        return $this->nombre;
    }
    public function getFrases(){
        return $this->frases;
    }
    public function setId($id){
        $this->id = $id;
    }
    public function setNombre($nombre){
        $this->nombre = $nombre;
    }
    public function setFrases($frases){
        $this->frases = $frases;
    }
}

